plugins/scheduled_task/lib/object_task_engine/KObjectTaskModifyEntryEngine.php

```diff
@@ -0,0 +1,217 @@
+	{
+		$entryId = $entryResultForMetadataUpdate->id;
+		
+		try
+		{
+			$metadataInputResult = $metadataPlugin->metadata->listAction($metadataFilter, null);
+		}
+		catch(Exception $e)
+		{
+			KalturaLog::notice("problem with metadata list with entry id $entryId - " . $e->getMessage());
+		}
+		
+		$tepmlateXmlObj = $this->getMetadataXmlTemplate($metadataPlugin, $outputMetadataProfileId);
+	
+		if($metadataInputResult && $metadataInputResult->totalCount != 0)
+		{
+			$metadataId = $metadataInputResult->objects[0]->id;
+			$metadataInputXmlStr = $metadataInputResult->objects[0]->xml;
+			$currentXmlObj = new SimpleXMLElement($metadataInputXmlStr);
+			
+			if(!is_null($tepmlateXmlObj))
+			{
+				$xmlData = $this->getUpdatedMetadataXmlStrFromEntry($entryResultForMetadataUpdate, $tepmlateXmlObj, $currentXmlObj, $outputMetadataArr);
+			
+				try
+				{
+					$metadataPlugin->metadata->update($metadataId, $xmlData, null);
+				}
+				catch(Exception $e)
+				{
+					KalturaLog::notice("problem with metadata update entry id $entryId - " . $e->getMessage());
+				}
+			}
+		}
+		else
+		{
+			
+			$xmlObj = new SimpleXMLElement("<metadata></metadata>");
+			$xmlData = $this->getUpdatedMetadataXmlStrFromEntry($entryResultForMetadataUpdate, $tepmlateXmlObj, $xmlObj, $outputMetadataArr);
+	
+			try
+			{
+				$metadataPlugin->metadata->add($outputMetadataProfileId, KalturaMetadataObjectType::ENTRY, $entryId, $xmlData);
+			}
+			catch(Exception $e)
+			{
+				KalturaLog::notice("problem with metadata add entry id $entryId - " . $e->getMessage());
+			}
+		}
+	}
+	
+	private function getUpdatedMetadataXmlStrFromEntry(KalturaBaseEntry $entryResultForMetadataUpdate, SimpleXMLElement $emptyXmlObj, SimpleXMLElement $currentXmlObj, array $outputMetadataArr)
+	{
+		$tempXmlObj = $emptyXmlObj;		
+		
+		KalturaLog::debug("currentXmlObj - " . print_r($currentXmlObj, true));
+		KalturaLog::debug("outputMetadataArr - " . print_r($outputMetadataArr, true));
+		
+		foreach($tempXmlObj as $metadataFieldName => $emptyXmlObjItem)
+		{
+			foreach($currentXmlObj as $currMetadataFieldName => $currentXmlObjItem)
+			{
+				if($currMetadataFieldName == $metadataFieldName && (string)$currentXmlObjItem != '')
+				{
+					$tempXmlObj->$metadataFieldName = $currentXmlObjItem;	
+					continue 2;
+				}				
+			}
+			
+			foreach($outputMetadataArr as $outputMetadataArrItem)
+			{
+				if($outputMetadataArrItem->key == $metadataFieldName)
+				{
+					$entryFieldName = $outputMetadataArrItem->value;
+					$emptyXmlObjItem = (string)$entryResultForMetadataUpdate->$entryFieldName;
+					$tempXmlObj->$metadataFieldName = (string)$entryResultForMetadataUpdate->$entryFieldName;
+					break;
+				}
+			}
+		}
+		return $tempXmlObj->asXml();
+	}
+	
+	private function updateEntryFromMetadata(&$metadataPlugin, array $inputMetadataArr, KalturaBaseEntry $entryObj, $metadataFilter)
```

Comment:
Objects are always passed by reference, remove `&` and add the type of the object on method's signature